.github/workflows/build_and_test_host_gcc.yml

```diff
@@ -15,22 +15,50 @@
 # This workflow runs Bazel build and test when triggered by specific pull request events.
 
 name: Bazel Build & Test communication module (with host toolchain GCC12.2)
+
 on:
   pull_request:
     types: [opened, reopened, synchronize]
   merge_group:
     types: [checks_requested]
+
 jobs:
-    build_and_test_gcc_host_gcc12:
-      runs-on: ubuntu-latest
-      container:
-        image: ghcr.io/eclipse-score/devcontainer:latest
-      steps:
-        - name: Checkout repository
-          uses: actions/checkout@v4.2.2
-        - name: Bazel build communication targets
-          run: |
-            bazel build //...
-        - name: Bazel test communication targets
-          run: |
-            bazel test //... --build_tests_only
+  build_and_test_gcc_host_gcc12:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/eclipse-score/devcontainer:latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4.2.2
+
+      - name: Set up Java 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: '17'
+
+
+      - name: Set up Bazel
+        uses: bazelbuild/setup-bazel@v5
+        with:
+          bazelisk: true
+
+  
+      - name: Bazel cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cache/bazel
+            ~/.cache/bazelisk
+          key: bazel-${{ runner.os }}-${{ hashFiles('WORKSPACE*', 'MODULE.bazel', '.bazelversion', '**/*.bzl', '**/*BUILD*') }}
+          restore-keys: |
+            bazel-${{ runner.os }}-
```

Comment:
We strongly believe that already available open source actions shall be used for this:

```
 uses: bazel-contrib/setup-bazel@0.15.0
        with:
          bazelisk-cache: true
          disk-cache: ${{ github.workflow }}
          repository-cache: true
```